android-components/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -35,58 +35,64 @@ class MediaSessionFullscreenFeature(
             flow.map {
                 it.tabs + it.customTabs
             }.map { tab ->
-                tab.filter { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
-            }.distinctUntilChanged().collect { states ->
-                processFullscreen(states)
-                processDeviceSleepMode(states)
+                tab.firstOrNull { it.mediaSessionState?.fullscreen == true }
+            }.distinctUntilChanged { old, new ->
+                old.hasSameOrientationInformationAs(new)
+            }.collect { state ->
+                /* there should only be one fullscreen session */
+                if (state == null) {
+                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
+                    activity.window?.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+                    return@collect
+                }
+
+                if (store.state.findCustomTabOrSelectedTab(tabId)?.id == state.id) {
+                    setOrientationForTabState(state)
+                }
```

Comment:
AFAICT this condition should always be true, since a tab can't be full-screen unless it's the active tab, right? I'm wondering if perhaps we want to capture/log an error in the `else` case, or if there's anything else we can to do indicate that we've reach (what should be) an impossible state. 

Mostly just sharing as food-for-thought and to check my own understanding. If you feel like there's anything we can improve here I think it would be fine for us to do it in a follow-up.

Reply:
It can be false, in the following case: 
Open Fenix, navigate to Youtube, play a video in fullscreen.
Put the video in PiP mode (pressing Home button or equivalent). 

Then open another app, that uses Custom Tabs. Fenix must be the default browser, and I think Chrome needs to be uninstalled (or desactivated). 
When the custom tab is opened, this code is reached and the condition is not met (the tab playing the video is not the active one).

I just checked because I was not totally sure, so thanks for raising this concern!

Reply:
Oh, good find! Yeah this logic definitely all checks out for me then

Reply:
Actually, I have one more question. I don't think we need to block on this, I'm just curious and having some trouble getting PiP to work in browser on youtube to check for myself.

> I'd say it's mainly because it is not technically possible to have two tabs in fullscreen at the same time. Does it seem to make sense to you?

Does a PiP video count as fullscreen? Would it then be possible to have a PiP video and a full screen video playing simultaneously? I'm mostly wondering where in the code we would actually prevent something like that